src/lib/utils/swap.utils.ts

```diff
@@ -76,36 +73,68 @@ export const getSwapTransferParams = async (
 
   const { approve, revoke } = await getTransferPermissions(
     tezos,
-    isSwapChains(chains) ? ROUTE3_CONTRACT : LIQUIDITY_BAKING_PROXY_CONTRACT,
+    isSwapHops(chains) ? ROUTE3_CONTRACT : LIQUIDITY_BAKING_PROXY_CONTRACT,
     accountPkh,
     fromRoute3Token,
     inputAmountAtomic
   );
 
   resultParams.unshift(...approve);
+  try {
+    const estimations = await tezos.estimate.batch(
+      resultParams.map(params => ({ kind: OpKind.TRANSACTION, ...params }))
+    );
+    estimations.forEach(({ suggestedFeeMutez, storageLimit, gasLimit }, index) => {
+      const currentParams = resultParams[index];
+      const approximateInternalOperationsCount = isSwapTransaction(currentParams)
+        ? 1 + (isSwapHops(chains) ? chains.hops.length : chains.xtzHops.length + chains.tzbtcHops.length)
+        : 1;
+      const gasCap = approximateInternalOperationsCount * GAS_CAP_PER_INTERNAL_OPERATION;
+      currentParams.fee = suggestedFeeMutez + Math.ceil(gasCap * FEE_PER_GAS_UNIT);
+      currentParams.storageLimit = storageLimit;
+      currentParams.gasLimit = gasLimit + gasCap;
+    });
+  } catch (e) {
+    console.error(e);
+    const swapOpParams = resultParams.find(isSwapTransaction)!;
+    // The quotients for calculations below are taken from analyzing several estimations
+    swapOpParams.fee = isSwapHops(chains)
+      ? 1100 + 1400 * chains.hops.length
+      : 4700 + 3900 * chains.tzbtcHops.length + 1200 * chains.xtzHops.length;
+    swapOpParams.gasLimit = isSwapHops(chains)
+      ? 8000 + 14000 * chains.hops.length
+      : 44000 + 39000 * chains.tzbtcHops.length + 12000 * chains.xtzHops.length;
```

Comment:
don't those values change over time? maybe just skip this hardcoding as you did on mobile?